spec/ui_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'console'
+require 'ui'
+
+describe UI do
+
+  let(:input)  {StringIO.new}
+  let(:output) {StringIO.new}
+  let(:ui)     {UI.new(Console.new(input, output))}
+
+  it "reads an answer" do
```

Comment:
The name of this test tells me that something regarding reading an answer will be tested. But the test verifies two different responsibilities of `ask_for_another`.
One is that it displays something to the UI, and the other one is that it returns a value that a user enters via stdin.
(This is true for the majority of tests in here)